src/Analytics/BasicProvider.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Message\Mothership\CMS\Analytics;
+
+/**
+ * @author Samuel Trangmar-Keates
+ *
+ * A basic provider class which is constructed with the reference, params and names
+ * required. Useful for basic providers.
+ *
+ * If there is a more complecated provider, then AnalyticsProviderInterface should
+ * be implemented
+ */
+class BasicProvider implements AnalyticsProvidorInterface
+{
+	/**
+	 * @var string
+	 */
+	private $_name;
+
+	/**
+	 * @var string
+	 */
+	private $_viewRef;
+
+	/**
+	 * @var array
+	 */
+	private $_params;
+
+	/**
+	 * Construct the object with the correct fields.
+	 * 
+	 * @param string $name    The object identifier
+	 * @param string $viewRef The view reference
+	 * @param array  $params  The parameters to pass to the view
+	 */
+	public function __construct($name, $viewRef, $params = [])
```

Comment:
Well I mean you would lose control of the constructor because, if you had a GoogleAnalyticsProvider, and a few releases later you decide that you need to load something so you pass it a loader, you would at that point need to make a whole new class rather than have it there already.